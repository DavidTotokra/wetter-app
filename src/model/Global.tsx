import styled from 'styled-components';
import { createGlobalStyle } from 'styled-components';

interface ContentProps {


}

export const Content = styled.div<ContentProps>`
  height: 100vh;
  display: flex;
  align-items: stretch;
  background-image: url(https://images.unsplash.com/photo-1561484930-998b6a7b22e8?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1170&q=80);
`;

export const GlobalStyle = createGlobalStyle`
  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }

  html {
    font-size: 62.5%;
  }

  body, input, textarea, button {
    font-family: 'Inter', sans-serif;
    font-weight: 400;
    color: #FFF;
  }

  h1, h2, h3, h4, h5, h6, strong {
    font-weight: 600;
  }

  button {
    cursor: pointer;
    border: 0;
    outline: 0;
  }

  input {
    outline: 0;
    border: 0;
  }

  a {
    text-decoration: none;
  }

  /* ::-webkit-scrollbar{
    width: 13px;
    height: 13px;
  }

  ::-webkit-scrollbar-thumb{
    background: rgba(255, 255, 255, 0.25);
  }

  ::-webkit-scrollbar-thumb:hover{
    background: rgba(255, 255, 255, 0.30);
  }

  ::-webkit-scrollbar-track{
    background: transparent;
    box-shadow: inset 0px 0px 0px 0px #F0F0F0;
  } */

  //// Media Queries
  html{
    @media(max-width: 1680px){
      font-size: 57.5%;
    }
  }
`;