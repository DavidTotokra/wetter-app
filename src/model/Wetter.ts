export interface OrtsAPI {
  address: {
    name: string;
    state: string;
    country: string;
  }
  display_name: string;
  lat: string;
  lon: string;
}

export interface Place {
  name: string;
  display_name?: string;
  lat: string;
  lon: string;
}


interface Wetter {
  icon: string;
  description: string;
}

interface TagWetter {
  rain?: number;
  temp: {
    min: number;
    max: number;
  }
  weather: Wetter[];
  sunrise: number;
}


export interface WetterData {
  current: {
    temp: number;
    clouds: number;
    humidity: number;
    wind_speed: number;
    rain?: {
      '1h': number;
    };
    weather: Wetter[];
  }
  daily: TagWetter[];
  timezone: 'UTC';
  place: Place;
}