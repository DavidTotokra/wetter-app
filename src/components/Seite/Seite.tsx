
import { Suche } from '../Suchen/Suchen';
import { Spinner } from '../Spinner/Spinner'

import { Place, WetterData } from '../../model/Wetter'

import { Content } from "./style";

export interface SeiteProps {
  setCurrentPlace: (params: Place) => void;
  wetterData: WetterData;
  loading: boolean;
  error: boolean;
}

 export function Seite(props:SeiteProps) {

  const setCurrentPlace = props.setCurrentPlace ;
  const wetterData = props.wetterData;
  const loading = props.loading;
  const error = props.error ; 
  






  function datum(index: number) {
    const date = new Date(); 

    const currentDay = date.getDate();
    const daysInCurrentMonth = new Date(date.getFullYear(), date.getMonth(), 0).getDate();
    const rest = (currentDay + index + 1) - daysInCurrentMonth;
    const actualDay = rest >= 0 ? rest + 1 : (currentDay + index + 1);

    const currentWeekday = date.toLocaleString('de-DE', { weekday: "long" });    
    const weekdays = ['Son', 'Mon', 'Diens', 'Wed', 'Donn', 'Frei', 'Sams', 'Son', 'Mon', 'Diens', 'Wed', 'Donn', 'Frei', 'Sams', 'Son', 'Mon'];
    const weekdayIndex = weekdays.findIndex(day => day === currentWeekday);
    const actualWeekday = weekdays[weekdayIndex + index + 6];
    return {actualDay, actualWeekday}
  }

  return (
    <Content>
      <Suche setCurrentPlace={setCurrentPlace} />

      <div className="separator"></div>

      <h1 className="heading">Wetter Details</h1>

      {loading ? <Spinner /> : !error && (
        <div className="container-flex-dual">
          <div>
            <span>Temperatur</span>
            <span>{Math.round(wetterData.current.temp)}°C</span>
          </div>
          <div>
            <span>Bewölkt</span>
            <span>{wetterData.current.clouds}%</span>
          </div>
          <div>
            <span>Luftfeuchtigkeit</span>
            <span>{wetterData.current.humidity}%</span>
          </div>
          <div>
            <span>Wind</span>
            <span>{(wetterData.current.wind_speed * 3.6).toFixed(1)}km/h</span>
          </div>
          <div>
            <span>Regen</span>
            <span>{wetterData.current.rain?.['1h'] || '0'}mm</span>
          </div>
        </div>
      )}

      {!loading && !error && (
        <>

          <div className="separator"></div>

          <h1 className="heading">Kommende Tage</h1>

          <div className="container-flex-column">
            {wetterData.daily.map((data, index) => {
              const { actualDay, actualWeekday  } = datum(index);
              return (
                <div key={data.sunrise}>
                  <div className="date-box">
                    <span>{actualDay}</span>
                    <span>{actualWeekday}</span>
                  </div>
                  <div className="temperatures">
                    <div>
                      <i className="fas fa-long-arrow-alt-down"></i>
                      <span>{Math.round(data.temp.max)}°</span>
                    </div>
                    <div>
                      <i className="fas fa-long-arrow-alt-up"></i>
                      <span>{Math.round(data.temp.min)}°</span>
                    </div>
                  </div>
                  <div className="rain">
                    <i className="fas fa-tint"></i>
                    <span>{data.rain? data.rain.toFixed(1) : '0'}mm</span>
                  </div>
                  <p className="description">{data.weather[0].description}.</p>
              </div>
              );
            })}
          </div>
        </>
      )}

    </Content>
  );
}