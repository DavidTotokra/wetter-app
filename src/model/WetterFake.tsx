import { WetterData } from "./Wetter";

export const  wetterData : WetterData={
    current: {
        temp: 0,
        clouds: 0,
        humidity: 0,
        wind_speed: 0,
        rain: undefined,
        weather: []
    },
    daily: [],
    timezone: "UTC",
    place: {
        name: 'Abidjan',
  
  lat: "5.345317",
  lon: "-4.024429",
    },

}