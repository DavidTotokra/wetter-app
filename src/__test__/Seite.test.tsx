import { act, render} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Seite } from '../components/Seite/Seite';
import {wetterData} from "../model/WetterFake";
import { Place } from '../model/Wetter';
import { componentRenderByMemoryRouter } from '../utils/testUtils';
import { Content } from '../components/Seite/style';


 


describe('Test CountryDetails Componet', () => {
    

    it('Should render correctly', () => {
        const Seit= render(<Content />)
        expect(Seit.container).toMatchSnapshot()
      })
      
    



    test('should render country info', async () => {
        // eslint-disable-next-line testing-library/no-unnecessary-act
        await act(async () => {
            componentRenderByMemoryRouter('/details/BD', <Seite setCurrentPlace={function (params: Place): void {
                throw new Error('Function not implemented.');
            } } wetterData={wetterData} loading={false} error={false} />);
        });
       
    });


});