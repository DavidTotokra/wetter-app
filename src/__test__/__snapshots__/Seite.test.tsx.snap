// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CountryDetails Componet Should render correctly 1`] = `
<div>
  <aside
    class="sc-dkzDqf QsDSv"
  />
</div>
`;
